pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -461,6 +465,47 @@ async def publish_client(self, working_dir, from_release_tag, release_name, buil
         # Publish the clients to our S3 bucket.
         await exectools.cmd_assert_async(f"aws s3 sync --no-progress --exact-timestamps {base_to_mirror_dir}/{build_arch} s3://art-srv-enterprise/pub/openshift-v4/{build_arch}", stdout=sys.stderr)
 
+    def publish_baremetal_installer_binary(self, from_release_tag: str, client_mirror_dir: str):
+        release_pullspec = f'{constants.QUAY_RELEASE_REPO_URL}:{from_release_tag}'
+        image_stat, baremetal_installer_pullspec = get_release_image_pullspec(release_pullspec, 'baremetal-installer')
+
+        if image_stat:  # != 0, image does not exist
+            self._logger.error('Error getting baremetal-installer binary from %s', release_pullspec)
+            return
+
+        # Get baremetal image pullspec
+        res, image_info = get_release_image_info_from_pullspec(baremetal_installer_pullspec)
+        if res:
+            self._logger.error('Error retrieving image info for %s', baremetal_installer_pullspec)
+            return
+
+        # Check rhel version (used for archive naming
+        display_name = image_info['config']['config']['Labels']['io.k8s.display-name']
+        if 'RHEL 8' in display_name:
+            rhel_version = 'rhel8'
+        else:
+            rhel_version = 'rhel9'
+
+        # oc adm release extract --command=openshift-baremetal-install -n=ocp <release-pullspec>
+        res, _ = extract_baremetal_installer(release_pullspec, client_mirror_dir)
+        if res:
+            self._logger.error('Error extracting baremetal-installer binary from %s', release_pullspec)
+            return
```

Comment:
Same here.

Reply:
Same as above: we need to exit from `publish_baremetal_installer_binary` if `oc adm release extract` fails

Reply:
Fixed with https://github.com/openshift-eng/aos-cd-jobs/pull/3774/commits/113c9a074dbe1daf09aa37634fff6cdb72f5e5e4